### PR TITLE
fix(GuildChannel): don't force parentID/permissionOverwrites to empty on create

### PR DESCRIPTION
**Pull request by MattIPv4**
**Please describe the changes this PR makes and why it should be merged:**

In #5796 a change was made that forced `parentID` to be null and `permissionOverwrites` to be an empty collection on all GuildChannels when they are created. This PR changes that logic to only apply those defaults if they're not set by the initial patch during the super call.

This should resolve the issue of channels firing an update event with parentID changing when they're patched by an event for the first time since the client starts.

Kudos to [muchnameless](https://discord.com/channels/222078108977594368/682166281826598932/853036819264569364) for spotting this before I could.

**Status and versioning classification:**

- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating

### CODE REVIEW

File: src/structures/GuildChannel.js [hunk 1]
```diff
@@ -36,8 +36,8 @@ class GuildChannel extends Channel {
      */
     this.guild = guild;
 
-    this.parentID = null;
-    this.permissionOverwrites = new Collection();
+    this.parentID = this.parentID || null;
+    this.permissionOverwrites = this.permissionOverwrites || new Collection();
```
Comment (kyranet) [inline src/structures/GuildChannel.js:40]:
```suggestion
    this.parentID ??= null;
    this.permissionOverwrites ??= new Collection();
```

**Review by kyranet — CHANGES_REQUESTED** [review]
Otherwise LGTM.

**Review by kyranet — APPROVED** [review]

File: src/structures/GuildChannel.js (hunk 1 above)
Comment (MattIPv4) [reply src/structures/GuildChannel.js]: I'd also originally written `||=` when I made this change locally and eslint got angry at me -- Node version bump when:tm:?

Comment (kyranet) [reply src/structures/GuildChannel.js]: We'll probably bump Node.js at the end of the year (around December) for Node.js v16 to become LTS and give users a grace period to upgrade if they didn't by that time.

**Review by iCrawl — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]

**Pull request merged**